### PR TITLE
pass annotations for reference fields to the EntityRef kotlin class. …

### PR DESCRIPTION
**Pull request by galganif**
Also adds a canonical @hardRef annotation.

Since I wanted to use the createAnnotations function from plan-generator.ts. I moved it to annotation2proto.ts and renamed this to annotations-utils.ts (let me know if I should split this in a separate pr).

### CODE REVIEW

File: src/runtime/tests/manifest-parser-test.ts [hunk 1]
```diff
@@ -71,6 +71,20 @@ describe('manifest parser', () => {
         h1: create 'my-id' @tiedToRuntime
         h2: create #mytag @tiedToArc`);
   });
+  it('parses schema annotations', () => {
+    parse(`
+      schema Abcd
+        hardRef: &Package @hardRef
```
Comment (mmandlis) [inline src/runtime/tests/manifest-parser-test.ts:77]: what is`@Package` annotation? i can't find where it is defined.

File: src/runtime/canonical-manifest.ts [hunk 2]
```diff
@@ -82,5 +82,10 @@ annotation policy(name: Text)
   retention: Source
   doc: 'Indicates that the target recipe should comply with the policy of the given name.'
 
+annotation hardRef
+  targets: [SchemaField]
+  retention: Source
+  doc: 'It can be used on reference fields: a hard reference indicate that the entity with that field should be deleted when the referenced entity is deleted.'
```
Comment (mmandlis) [inline src/runtime/canonical-manifest.ts:88]: nit - s/indicate/indicates

File: java/arcs/core/data/SchemaFields.kt [hunk 3]
```diff
@@ -23,7 +23,10 @@ sealed class FieldType(
     }
 
     /** A reference to an entity. */
-    data class EntityRef(val schemaHash: String) : FieldType(Tag.EntityRef) {
+    data class EntityRef(
```
Comment (mmandlis) [inline java/arcs/core/data/SchemaFields.kt:26]: maybe better to put the annotations list in `FieldType` directly? the manifest supports annotations on any field, so it's more consistent.

File: src/runtime/tests/manifest-parser-test.ts (hunk 1 above)
Comment (galganif) [reply src/runtime/tests/manifest-parser-test.ts]: changing the annotation to @aFoo to make it clear it is not a special annotation defined anywhere

File: src/runtime/canonical-manifest.ts (hunk 2 above)
Comment (galganif) [reply src/runtime/canonical-manifest.ts]: done

File: java/arcs/core/data/SchemaFields.kt (hunk 3 above)
Comment (galganif) [reply java/arcs/core/data/SchemaFields.kt:27]: done

**Pull request merged**